src/zenml/zen_stores/secrets_stores/base_secrets_store.py

```diff
@@ -58,6 +59,79 @@ class BaseSecretsStore(BaseModel, SecretsStoreInterface, ABC):
     TYPE: ClassVar[SecretsStoreType]
     CONFIG_TYPE: ClassVar[Type[SecretsStoreConfiguration]]
 
+    @model_validator(mode="before")
+    @classmethod
+    @before_validator_handler
+    def convert_config(cls, data: Dict[str, Any]) -> Dict[str, Any]:
+        """Method to infer the correct type of the config and convert.
+
+        Args:
+            data: The provided configuration object, can potentially be a
+                generic object
+
+        Raises:
+            ValueError: If the provided config object's type does not match
+                any of the current implementations.
+
+        Returns:
+            The converted configuration object.
+        """
+        if data["config"].type == SecretsStoreType.SQL:
+            from zenml.zen_stores.secrets_stores.sql_secrets_store import (
+                SqlSecretsStoreConfiguration,
```

Comment:
Again, why we run all those casting below? Is it P2 limitation or what?

Reply:
Same as above.